docs/userguide/development_mode.rst

```diff
@@ -1,34 +1,150 @@
+
+   .. code-block:: bash
+
+       sudo apt install python3-venv
+
+   Alternatively, you can also try installing :pypi:`virtualenᴠ`.
+   More information is available on the Python Packaging User Guide on
+   :doc:`PyPUG:guides/installing-using-pip-and-virtual-environments`.
+
+.. note::
+    .. versionchanged:: v63.0.0
+       Editable installation hooks implemented according to :pep:`660`.
+       Support for :pep:`namespace packages <420>` is still **EXPERIMENTAL**.
+
+
+"Strict" editable installs
+--------------------------
+
+When thinking about editable installations, users might have the following
+expectations:
+
+1. It should allow developers to create add new files and have them
+   automatically exposed.
+2. It should behave as close as possible to a regular installation and help
+   users to detect problems (e.g. new files not being included in the distribution).
+
+Unfortunately these expectations are in conflict with each other.
+To solve this problem ``setuptools`` allows developers to choose a more
+*"strict"* mode for the editable installation. This can be done by passing
+a special *configuration setting* via :pypi:`pip`, as indicated bellow:
+
+.. code-block:: bash
+
+    pip install -e . --config-settings editable_mode=strict
+
+In this mode, new files **won't** be exposed and the editable installs will
+try to mimic as much as possible the behavior of a regular install.
+Under the hood, ``setuptools`` will create a tree of file links in an auxiliary
+directory (``$your_project_dir/build``) and add it to ``PYTHONPATH`` via a
+:mod:`.pth file <site>`. (Please be careful to not delete this repository
+by mistake otherwise your files may stop being accessible).
+
+
+.. note::
+    .. versionadded:: v63.0.0
+       *Strict* mode implemented as **EXPERIMENTAL**.
+
+
+Limitations
+-----------
+
+- The *editable* term is used to refer only to Python modules
+  inside the package directories. Non-Python files, external (data) files,
+  executable script files, binary extensions, headers and metadata may be
+  exposed as a *snapshot* of the version they were at the moment of the
+  installation.
+- Adding new dependencies or entry-points to your project require
+  a fresh "editable" re-installation.
+- Console scripts and GUI scripts **MUST** be specified via entry-points
+  to work properly.
+- *Strict* editable installs require the file system to support
+  either :wiki:`symbolic <symbolic link>` or :wiki:`hard links <hard link>`.
+- Editable installations may not work with
+  :doc:`namespaces created with pkgutil or pkg_resouces
+  <PyPUG:guides/packaging-namespace-packages>`.
+  Please use :pep:`420`-style implicit namespaces.
+
+.. attention::
+   Editable installs are **not a perfect replacement for regular installs**
+   in a test environment. When in doubt, please test your projects as
+   installed via a regular wheel. There are tools in the Python ecosystem,
+   like :pypi:`tox` or :pypi:`nox`, that can help you with that.
```

Comment:
Just a minor point, but doesn't tox use `sdist` by default and rely on [a plugin](https://github.com/ionelmc/tox-wheel) for wheel?

Reply:
I think you don't exactly need a plugin to achieve that with tox... 

It seems that the default behaviour is to build an sdist file and then use `pip install`... If that is the case, `pip install` will internally build a wheel in the final step before installing anyway. (The sdist creation can be forced to follow PEP 517 with the [isolated_build](https://tox.wiki/en/latest/config.html?highlight=isolated_build#conf-isolated_build) config).

Additionally,  a developer could use:

- [install_command](https://tox.wiki/en/latest/config.html?highlight=isolated_build#conf-install_command) config.

- `--installpkg` CLI flag, pointing to a wheel.

Maybe it would be good to confirm this behaviour with the tox developers.

Do you have any suggestions?

Reply:
Just complementing some words like `tox (with appropriate configuration) and nox` would fine for me.

BTW, what is an effective way to understand the difference between editable/wheel install, after I tried `pip install .` in my testing virtualenv?

Reply:
Thank you very much for the suggestions @kngwyu! I will add those in a follow up review.

> BTW, what is an effective way to understand the difference between editable/wheel install, after I tried pip install . in my testing virtualenv?

Let's say your are creating a `mypkg` with a `__main__.py` module like the following:

```python
# mypkg/__main__.py
print("hello world")
```

When you do `pip install .`, pip will create a wheel that will correspond to a "snapshot in time" of your package. So if later you change the print statement to `print("hello universe")`, your package will still display the old message:

```bash
$ cd /tmp  # let's change the directory to avoid messing with PYTHONPATH
$ python -m mypkg
hello world
```

If you do an editable install instead `pip install -e .`, then it would display the updated message:

```bash
$ cd /tmp
$ python -m mypkg
hello universe
```

If you feel like the docs could improve, please feel free to add suggestions, they are always welcome!